testing/fix_coverage.py

```diff
@@ -1,37 +1,31 @@
-import os.path
-
-from coverage.data import CoverageData
-
-
-def merge_coverage(coverage_data, from_path, to_path):
-    for filename in coverage_data.measured_files():
-        result_filename = filename.split(from_path)[-1]
-        if filename == result_filename:
-            continue
-
-        result_filename = result_filename.lstrip('/')
-        result_filename = os.path.join(to_path, result_filename)
-        result_filename = os.path.abspath(result_filename)
-        if os.path.exists(result_filename):
-            coverage_data.add_arcs(
-                {result_filename: coverage_data.arcs(filename)}
-            )
-
-        del coverage_data._arcs[filename]
-        coverage_data._validate_invariants()
-
-
-def fix_coverage(from_path, to_path):
-    coverage_data = CoverageData()
-    os.rename('.coverage', '.coverage.orig')
-    coverage_data.read_file('.coverage.orig')
-    merge_coverage(coverage_data, from_path, to_path)
-    coverage_data.write_file('.coverage')
-    os.remove('.coverage.orig')
+import sqlite3
+
+QUERY_SITE_PACKAGES = '''\
+SELECT id FROM file WHERE path LIKE '%site-packages/future_fstrings.py'
+'''
+QUERY_NOT_SITE_PACKAGES = '''\
+SELECT id FROM file
+WHERE (
+    path LIKE '%/future_fstrings.py' AND
+    path NOT LIKE '%site-packages/future_fstrings.py'
+)
+'''
+QUERY_TEST = '''\
+SELECT id FROM file WHERE path LIKE '%/future_fstrings_test.py'
+'''
+MERGE_FILE_IN_ARC = 'UPDATE arc SET file_id = ? WHERE file_id = ?'
+DELETE_FROM_ARC = 'DELETE FROM arc WHERE file_id NOT IN (?, ?)'
+DELETE_FROM_FILE = 'DELETE FROM file WHERE id NOT IN (?, ?)'
```

Comment:
@nedbat any advice on this? I don't love it but it works and the old api doesn't work any more :(

Reply:
I have to admit that I am confused what the larger purpose of this code is.  I have a feeling we started a conversation about it, but I don't remember.

Reply:
heheh yeah a while back

this package is admittedly very weird and I had to do some tricks to make coverage even somewhat play nicely with it (since the encoding hack happens way too early in the process for even coverage to notice it)

so what I currently do is:

1. trace `future_fstrings` in site-packages
2. [`reload(future_fstrings)` at the end of the tests](https://github.com/asottile/future-fstrings/blob/1b9218fd8fb152c262319ccce26280a9225a80ae/tests/future_fstrings_test.py#L201-L207)
3. merge together the site-packages tracing into the source directory

Reply:
It seems like something the `[paths]` configuration section could handle?  What's an example of the wrong paths before the fixup, and the right paths afterwards?

Reply:
before it's `.tox/py27/lib/python2.7/site-packages/future_fstrings.py` and `future_fstrings.py` and I merge those into `future_fstrings.py`